SYCL/Matrix/joint_matrix_query_use_default.cpp

```diff
@@ -0,0 +1,166 @@
+//==------ joint_matrix_query_use_default.cpp  - DPC++ joint_matrix---------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// REQUIRES: matrix
+
```

Comment:
Could you please add a comment clarifying what exactly is checked by this test. It is not clear which API is being tested.

Reply:
We recently updated the matrix API to introduce a new argument "use" in joint_matrix type.
https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/experimental/sycl_ext_oneapi_matrix/sycl_ext_oneapi_matrix.asciidoc
the old API was moved to https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/deprecated/sycl_ext_oneapi_matrix_no_use.asciidoc

The specific changes that I am making here are done in https://github.com/intel/llvm/pull/6981